src/NHibernate.Test/NHSpecificTest/NH1001/Fixture.cs

```diff
@@ -12,54 +12,217 @@ protected override void Configure(Configuration configuration)
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(1));
+			}
+		}
+
+		[Test]
+		public void Department2IsNotNull()
+		{
+			ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 99999 WHERE EMPLOYEE_ID = {employeeId}");
+
+			IStatistics statistics = Sfi.Statistics;
+			statistics.Clear();
+
+			using (ISession session = OpenSession())
+			{
+				var employee = session.Get<Employee>(employeeId);
+				Assert.That(employee.Department1, Is.Not.Null);
+				Assert.That(employee.Department2, Is.Null);
+				Assert.That(employee.Department3, Is.Not.Null);
+				Assert.That(employee.Address, Is.Not.Null);
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(1));
+			}
+		}
+
+		[Test]
+		public void AddressNull()
+		{
+			ExecuteStatement(
+				$"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 12, ADDRESS_ID = 99999 WHERE EMPLOYEE_ID = {employeeId}");
+
+			IStatistics statistics = Sfi.Statistics;
+			statistics.Clear();
+
+			using (ISession session = OpenSession())
+			{
+				var employee = session.Get<Employee>(employeeId);
+				Assert.That(employee.Department1, Is.Not.Null);
+				Assert.That(employee.Department2, Is.Not.Null);
+				Assert.That(employee.Department3, Is.Not.Null);
+				Assert.That(employee.Address, Is.Null);
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(1));
+			}
+		}
+
+		[Test]
+		public void PhoneIsNull()
+		{
+			ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 12, ADDRESS_ID = 15 WHERE EMPLOYEE_ID = {employeeId}");
+			ExecuteStatement($"UPDATE PHONES SET EMPLOYEE_ID = NULL");
+
+			IStatistics statistics = Sfi.Statistics;
+			statistics.Clear();
+
+			using (ISession session = OpenSession())
+			{
+				var employee = session.Get<Employee>(employeeId);
+				Assert.That(employee.Department1, Is.Not.Null);
+				Assert.That(employee.Department2, Is.Not.Null);
+				Assert.That(employee.Department3, Is.Not.Null);
+				Assert.That(employee.Address, Is.Not.Null);
+				Assert.That(employee.Phones, Is.Not.Null);
+				Assert.That(employee.Phones.Count, Is.Zero);
+				Assert.That(statistics.PrepareStatementCount, Is.EqualTo(1));
+			}
+		}
+		
+		[Test]
+		public void Department2IsNull_QueryOver()
+		{
+			ExecuteStatement($"UPDATE EMPLOYEES SET DEPARTMENT_ID_1 = 11, DEPARTMENT_ID_2 = 99999 WHERE EMPLOYEE_ID = {employeeId}");
+
+			IStatistics statistics = Sfi.Statistics;
+			statistics.Clear();
+
+			using (ISession session = OpenSession())
+			{
+				var employee = session.QueryOver<Employee>()
+					.Fetch(SelectMode.Fetch, e => e.Department2)
+					.Fetch(SelectMode.Skip, e => e.Department1, e => e.Department3, e => e.Address)
+					.Where(e => e.Id == employeeId)
+					.SingleOrDefault();
+
+				Assert.That(employee.Department1, Is.Not.Null);
+				Assert.That(employee.Department2, Is.Null);
+				Assert.That(employee.Department3, Is.Not.Null);
+				Assert.That(employee.Address, Is.Not.Null);
```

Comment:
In this test, how many statements do we expect to be executed? /cc @bahusoid

Reply:
My main point was to provide example when order of entities in result set is different from mapping (so mapping based index check is wrong).  IMHO number of queries are not relevant - but I can double check later if you think it's useful (just let me know).
Theoretically this fix should save 1 additional request for checking Department2 but all other `not-found` ignore properties should still trigger requests.
So I think 4 statements in total : 1 for main query + 3 for Department1, Department3, Address "not found" ignore check.

And regarding current fix. This Aliases parse logic doesn't feel reliable and looks like depends on some internal knowledge of how aliases are generated. I would avoid such dependencies.